src/cpp/core/DatabaseTests.cpp

```diff
@@ -0,0 +1,471 @@
+/*
+ * DatabaseTests.cpp
+ *
+ * Copyright (C) 2020 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include <tests/TestThat.hpp>
+
+#include <core/Database.hpp>
+#include <core/FileSerializer.hpp>
+#include <core/system/System.hpp>
+#include <shared_core/SafeConvert.hpp>
+
+#include <soci/boost-tuple.h>
+#include <soci/session.h>
+#include <soci/sqlite3/soci-sqlite3.h>
+
+namespace rstudio {
+namespace unit_tests {
+
+using namespace core;
+using namespace core::database;
+
+core::database::SqliteConnectionOptions sqliteConnectionOptions()
+{
+   return SqliteConnectionOptions { "/tmp/rstudio-test-db" };
+}
+
+core::database::PostgresqlConnectionOptions postgresConnectionOptions()
+{
+   PostgresqlConnectionOptions options;
+   options.connectionTimeoutSeconds = 10;
+   options.database = "rstudio-test";
+   options.host = "localhost";
+   options.user = "postgres";
+   options.password = "postgres";
+
+   return options;
+}
+
+TEST_CASE("Database", "[.database]")
+{
+   test_that("Test Setup")
+   {
+      // ensure that the test databases do not exist
+      FilePath sqliteDbPath("/tmp/rstudio-test-db");
+      sqliteDbPath.removeIfExists();
+
+      boost::shared_ptr<IConnection> connection;
+      Error error = connect(postgresConnectionOptions(), &connection);
+      if (error)
+         return;
```

Comment:
Yes, and that is intended - if the test database cannot be connected to, we won't run the tests. This is also okay because these tests are never run automatically.

Reply:
Yeah, that makes sense. I just wasn't sure if a return exits the whole test case or just the test_that section.